org.eclipse.winery.lsp/src/main/java/org/eclipse/winery/lsp/Server/ServerCore/DataModels/PropertyDefinition.java

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+/**
+ * Property Definition
+ * For more details on the TOSCA specification, visit:
+ * <a href="https://docs.oasis-open.org/tosca/TOSCA/v2.0/csd06/TOSCA-v2.0-csd06.html#93-property-definition">Property Definition</a>
+ */
+
+package org.eclipse.winery.lsp.Server.ServerCore.DataModels;
+
+import java.util.Map;
+import java.util.Optional;
+
+public record PropertyDefinition<T>(    String type,
+                                     Optional<String> description,
+                                     Optional<Map<String, Object>> metadata,
+                                     Optional<Boolean> required,
+                                     Optional<T> Default , //must match property type
```

Comment:
This is also a TODO.

Reply:
I did initially handle that Default match property type.